src/runtime/client-event-manager-test.js

```diff
@@ -34,272 +35,310 @@ const DEFAULT_EVENT = {
 };
 
 describes.sandboxed('EventManager', {}, () => {
-  describe('promises', () => {
-    let eventMan;
-    let resolver;
-    let rejector;
-    let counter;
+  describe('configuration', () => {
+    let eventManager;
+    let indicateConfigurationSucceeded;
+    let indicateConfigurationFailed;
+    let events;
+    let configurationPromise;
+
     beforeEach(() => {
-      eventMan = new ClientEventManager(
-        new Promise((resolve, reject) => {
-          rejector = reject;
-          resolver = resolve;
-        })
-      );
-      counter = 0;
-      eventMan.registerEventListener(() => counter++);
-      eventMan.logEvent(DEFAULT_EVENT);
-      //ensure it has not logged yet
-      expect(counter).to.equal(0);
+      configurationPromise = new Promise((resolve, reject) => {
+        indicateConfigurationSucceeded = resolve;
+        indicateConfigurationFailed = reject;
+      });
+      eventManager = new ClientEventManager(configurationPromise);
+      events = [];
+      eventManager.registerEventListener(e => events.push(e));
+    });
+
+    it('should not log events before configuration succeeds', async () => {
+      // Request event log.
+      eventManager.logEvent(DEFAULT_EVENT);
+
+      // Event should not be logged.
+      await tick();
+      expect(events).to.deep.equal([]);
     });
 
-    it('should not log events until its promise is resolved', async () => {
-      let counter2 = 0;
+    it('should log events after configuration succeeds', async () => {
+      // Request event log.
+      eventManager.logEvent(DEFAULT_EVENT);
+
+      indicateConfigurationSucceeded();
```

Comment:
We should do an await tick before indicating configuration success.  Then check to ensure no log was sent.  Then indicate configuration success.  Then await the logging promise (instead of tick which might be flaky).  Then check to ensure events === [DEFAULT_EVENT]

Reply:
This is covered in the test above: 'should not log events before configuration succeeds'